backend/kernelCI_app/views/treeDetailsSlowView.py

```diff
@@ -0,0 +1,202 @@
+from django.http import JsonResponse
+from django.views import View
+from kernelCI_app.utils import extract_error_message, extract_platform
+from django.db import connection
+
+
+class TreeDetailsSlow(View):
+    def get(self, request, commit_hash: str | None):
+        origin_param = request.GET.get("origin")
+        git_url_param = request.GET.get("git_url")
+        git_branch_param = request.GET.get("git_branch")
+        query = """
+        SELECT
+                tests.build_id AS tests_build_id,
+                tests.id AS tests_id,
+                tests.origin AS tests_origin,
+                tests.environment_comment AS tests_enviroment_comment,
+                tests.environment_misc AS tests_enviroment_misc,
+                tests.path AS tests_path,
+                tests.comment AS tests_comment,
+                tests.log_url AS tests_log_url,
+                tests.status AS tests_status,
+                tests.waived AS tests_waived,
+                tests.start_time AS tests_start_time,
+                tests.duration AS tests_duration,
+                tests.number_value AS tests_number_value,
+                tests.misc AS tests_misc,
+                builds_filter.*
+        FROM
+            (
+                SELECT
+                    builds.checkout_id AS builds_checkout_id,
+                    builds.id AS builds_id,
+                    builds.comment AS builds_comment,
+                    builds.start_time AS builds_start_time,
+                    builds.duration AS builds_duration,
+                    builds.architecture AS builds_architecture,
+                    builds.command AS builds_command,
+                    builds.compiler AS builds_compiler,
+                    builds.config_name AS builds_config_name,
+                    builds.config_url AS builds_config_url,
+                    builds.log_url AS builds_log_url,
+                    builds.valid AS builds_valid,
+                    tree_head.*
+                FROM
+                    (
+                        SELECT
+                            checkouts.id AS checkout_id
+                        FROM
+                            checkouts
+                        WHERE
+                            checkouts.git_commit_hash = %(commit_hash)s AND
+                            checkouts.git_repository_url = %(git_url_param)s AND
+                            checkouts.git_repository_branch = %(git_branch_param)s AND
+                            checkouts.origin = %(origin_param)s
+                    ) AS tree_head
+                LEFT JOIN builds
+                    ON tree_head.checkout_id = builds.checkout_id
+                WHERE
+                    builds.origin = %(origin_param)s
+            ) AS builds_filter
+        LEFT JOIN tests
```

Comment:
these joins can be inner joins because we are looking for tests, we don't have business with checkouts that does not have builds and builds that does not have tests